lib/TPP/ToBlockLayoutAndBack.cpp

```diff
@@ -482,6 +482,98 @@ struct PropagateThroughPadOp : public OpRewritePattern<tensor::PadOp> {
 // PropagateThroughElementWiseOp
 //===----------------------------------------------------------------------===//
 
+struct PackInfo {
```

Comment:
This is very different from https://reviews.llvm.org/D139680. We already have a big delta between upstream pack and unpack, can we keep it low by following what is being done upstream here too?

Reply:
the struct or the entire PR?

Reply:
For now just the struct. We need to converge on everything that we can, so that we can pull the upstream changes as soon as possible.

Reply:
I can align the name better but we have `SmallVector<int64_t> interchangeVector;` which is something not upstream. Also notice that this PR is for the unpack while upstream is for pack. Is this name-changing fine?

Reply:
Sorry, `DenseMap`

Reply:
> Could you please be more specific on what you are referring to?
> 
> `packedTileSize` -> `dimAndTileMapping` [same container] `outerDimsOnDomainPerm`->`tileLoopPerms` [same container]
> 
> `interchangeVector` is not in the upstream as it is not used.

Ah, you're right! Perhaps just a renaming should do it then.